src/Marten/Linq/SqlGeneration/Statement.cs

```diff
@@ -45,7 +45,7 @@ private set
         public string ExportName { get; protected set; }
 
 
-        public IList<Ordering> Orderings { get; protected set; } = new List<Ordering>();
+        public IList<(Ordering Ordering, bool CaseInsensitive)> Orderings { get; protected set; } = new List<(Ordering, bool)>();
```

Comment:
My first idea was to create a new type by inheriting from Ordering with the case insensitive flag. The Ordering class in Remotion.Linq is sealed so that was not possible.

Would it be better in the long run to replace the Ordering here with our own Ordering type in Marten? This would still be breaking change but might make it easier to add new flags in the future if need be.

Reply:
Some thoughts to handle the binary compatibility:
- Mark `Orderings` member as obsolete as below:
```cs
[Obsolete("Orderings is deprecated, please use OrderingsWithCaseSensitivity", true)]
public IList<Ordering> Orderings { get; protected set; } = new List<Ordering>();
```
- As you indicated we can have a new `OrderByClause` Type in Marten which has `Ordering` from `Remotion.Linq` as a property and also other properties such as case insensitivity. Use this as a new type to create a property as `public IList<OrderByClause> OrderByClauses { get; protected set; } = new List<OrderByClause>();`

With above, it does not break anything. Thoughts?

Alternatively, the other reviewers can take a call to accept the change "as is" taking into the breaking change since it is seldom used elsewhere outside Marten.

Reply:
I think that having two lists will make this a bit more complicated, but it might still be the best option.

The only problem I see with this is, that we need to try to keep the lists in sync. This is not a problem in Marten's code base as we only add items into the list and can easily add the ordering into both lists. However this gets a bit more interesting in case the current Orderings collection is modified by the consumers of Marten. I think that this can still be solved by using the current Orderings list as the primary source for orderings and use the new list as a lookup to check the casing flag.

This solution does not keep the lists in sync as the modifications done to the old list outside of Marten are not updated into the new collection. This however should not break the case where current list is modified outside of Marten.

Reply:
Setter is protected so it can't be updated externally outside Marten so we can add an empty block for it in terms of implementation. For the deprecated `Orderings` getter implementation, we can always compute the return value based on the new `OrderByClauses` to return the list of `Remotion.Linq.Ordering`.  

Within Marten code base, we can change the  usage everywhere to use `OrderByClauses.`. This way, everything is in sync.

Reply:
Let us wait for @jeremydmiller and @oskardudycz comments, given that it is very unlikely to be used outside Marten, we could probably acknowledge and take in the breaking change "as is".

Reply:
I need to look through things better, but I'd consider the Statement/LinqHandlerBuilder to be an internal detail even though the type itself is public (and has to stay that way for code generation as I recall)

Reply:
This will be released as part of 5.0.0